bezier/src/main/java/io/channel/bezier/compose/component/switch/source/BezierSwitchControl.kt

```diff
@@ -0,0 +1,201 @@
+private val SwitchPadding = 3.dp
+private val ThumbSize = 22.dp
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+internal fun BezierSwitchControl(
+        state: BezierSwitchControlState,
+        onCheckedChange: (Boolean) -> Unit,
+        modifier: Modifier = Modifier,
+) {
+    val coroutineScope = rememberCoroutineScope()
+
+    val checked = state.checked
+    val anchoredDraggableState = state.anchoredDraggableState
+
+    LaunchedEffect(checked, state.forceAnimationCheck) {
+        if (checked != anchoredDraggableState.currentValue) {
+            anchoredDraggableState.animateTo(checked)
+        }
+    }
+
+    Box(
+            modifier = modifier
+                    .padding(2.dp)
+                    .clip(CircleShape)
+                    .requiredSize(width = SwitchWidth, height = 28.dp)
+                    .clickable(
+                            interactionSource = remember { MutableInteractionSource() },
+                            indication = null,
+                            onClick = {
+                                coroutineScope.launch {
+                                    state.trySwitch(onCheckedChange)
+                                }
+                            },
+                    ),
+    ) {
+        Track(fraction = { anchoredDraggableState.requireOffset().dp / SwitchWidth })
+        Box(
+                modifier = Modifier
+                        .padding(SwitchPadding)
+                        .anchoredDraggable(
+                                state = anchoredDraggableState,
+                                orientation = Orientation.Horizontal,
+                        ),
+        ) {
+            Thumb(thumbValue = { anchoredDraggableState.requireOffset() })
+        }
+    }
+}
+
+/**
+ * 리컴포지션을 피하기 위해 값을 Canvas 가 읽도록 람다를 받습니다.
+ * [공식 문서](https://developer.android.com/develop/ui/compose/phases?hl=ko)
+ */
+@Composable
+private fun Track(fraction: () -> Float) {
+    val start = BezierTheme.colorSchemes.bgBlackDark.color
+    val stop = BezierTheme.colorSchemes.bgGreenNormal.color
+
+    Canvas(
+            modifier = Modifier
+                    .fillMaxSize(),
+    ) {
+        val backgroundColor = lerp(
+                start = start,
+                stop = stop,
+                fraction = fraction(),
+        )
+
+        drawRect(backgroundColor)
+    }
+}
+
+@Composable
+private fun Thumb(thumbValue: () -> Float) {
+    val color = BezierTheme.colorSchemes.fgAbsoluteWhiteDark.color
+
+    Canvas(
+            modifier = Modifier
+                    .size(ThumbSize)
+                    .bezierShadow(ShadowStyle.Shadow2, CircleShape),
+    ) {
+        drawLine(
+                color = color,
+                start = Offset(thumbValue() + (size.width / 2), this.center.y),
+                end = Offset(thumbValue() + (size.width / 2), this.center.y),
+                strokeWidth = size.width,
+                cap = StrokeCap.Round,
+        )
+    }
```

Comment:
thumb를 라인으로 그린다는게 어색할 수 있을거 같다는 생각이 들어서 그러는데 drawRoundRect로 하는 좋은 아이디어가 있는데 어떠신가요? 
```suggestion
    Canvas(
            modifier = Modifier
                    .size(ThumbSize)
                    .bezierShadow(ShadowStyle.Shadow2, CircleShape),
    ) {
        drawRoundRect(
                color = color,
                topLeft = Offset(thumbValue.invoke(), 0f),
                cornerRadius = CornerRadius(100f, 100f)
        )
```

Reply:
음 이렇게 구현하면 좀 다른부분을 바꿔야하는군요 흠흠